homework/index.js

```diff
@@ -16,31 +30,97 @@
     xhr.send();
   }
 
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
-      } else {
-        elem.setAttribute(key, value);
-      }
+  function renderRepositories(repo, ulRepoInfo) {
+    const url = `https://api.github.com/repos/HackYourFuture/${repo}`;
+    fetchJSON(url, (err, repoInfo) => {
+      ulRepoInfo.innerHTML = '';
+      const li = createAndAppend('li', ulRepoInfo, { id: 'li1' });
+      const table = createAndAppend('table', li);
+      const tbody = createAndAppend('tbody', table);
+
+      const tr1 = createAndAppend('tr', tbody);
+      createAndAppend('td', tr1, { text: 'Repository:', class: 'label' });
+      createAndAppend('td', tr1);
+      createAndAppend('a', tr1.lastChild, {
+        href: repoInfo.html_url,
+        target: '_blank',
+        text: repoInfo.name,
+      });
+
+      const tr2 = createAndAppend('tr', tbody);
+      createAndAppend('td', tr2, { text: 'Description:', class: 'label' });
+      createAndAppend('td', tr2, { text: `${repoInfo.description}` });
+
+      const tr3 = createAndAppend('tr', tbody);
+      createAndAppend('td', tr3, { text: 'Forks:', class: 'label' });
+      createAndAppend('td', tr3, { text: `${repoInfo.forks}` });
+
+      const tr4 = createAndAppend('tr', tbody);
+      createAndAppend('td', tr4, { text: 'Updated:', class: 'label' });
+      createAndAppend('td', tr4, { text: new Date(repoInfo.updated_at).toLocaleString() });
     });
-    return elem;
+  }
+
+  function renderContributors(repoName, ulContributorInfo) {
+    const url = `https://api.github.com/repos/HackYourFuture/${repoName}/contributors`;
```

Comment:
Do not contruct this url yourself. Use the `contributors_url` as mentioned in the assignment. GitHub may choose to alter the internal format of this url, but will not likely change the name of the property as this is the documented way to obtain the contributors information.